subscriptions/subscriptions-impl/src/main/java/com/duckduckgo/subscriptions/impl/billing/RealBillingClientAdapter.kt

```diff
@@ -75,26 +74,30 @@ class RealBillingClientAdapter @Inject constructor(
             .build()
 
         return suspendCoroutine { continuation ->
-            billingClient?.startConnection(
-                object : BillingClientStateListener {
-                    override fun onBillingServiceDisconnected() {
-                        disconnectionListener.invoke()
-                    }
-
-                    override fun onBillingSetupFinished(p0: BillingResult) {
-                        val result = when (p0.responseCode) {
-                            BillingResponseCode.OK -> Success
-                            else -> Failure(billingError = p0.responseCode.toBillingError())
+            try {
+                billingClient?.startConnection(
+                    object : BillingClientStateListener {
+                        override fun onBillingServiceDisconnected() {
+                            disconnectionListener.invoke()
                         }
 
-                        try {
-                            continuation.resume(result)
-                        } catch (e: IllegalStateException) {
-                            logcat(priority = WARN) { "onBillingSetupFinished() invoked more than once" }
+                        override fun onBillingSetupFinished(p0: BillingResult) {
+                            val result = when (p0.responseCode) {
+                                BillingResponseCode.OK -> Success
+                                else -> Failure(billingError = p0.responseCode.toBillingError())
+                            }
+
+                            try {
+                                continuation.resume(result)
+                            } catch (e: IllegalStateException) {
+                                logcat(priority = WARN) { "onBillingSetupFinished() invoked more than once" }
+                            }
                         }
-                    }
-                },
-            )
+                    },
+                )
+            } catch (e: SecurityException) {
+                logcat(priority = WARN) { "Exception when starting a connection to billing, ignoring" }
```

Comment:
Assuming the `SecurityException` is thrown by `startConnection()` before `onBillingSetupFinished()` listener is called, the coroutine that invokes `BillingClientAdapter::connect()` will be left hanging in suspended state indefinitely. I think we should call `continuation.resume(Failure(BILLING_CRASH_ERROR))` after catching the exception.